File: bindings/typescript/tools/docs-surface.mjs
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

const root = path.resolve(path.dirname(fileURLToPath(import.meta.url)), "..");
const snapshot = fs.readFileSync(path.join(root, "api/public-api.txt"), "utf8");
const symbols = snapshot
  .split("\n")
  .filter((line) => line && !line.startsWith("#"))
  .map((line) => {
    const [entrypoint, name, kind] = line.split("\t");
    if (!entrypoint || !name || !kind) throw new TypeError(`invalid public API line: ${line}`);
    return { entrypoint, name, kind };
  })
  .sort((left, right) => `${left.entrypoint}\0${left.name}`.localeCompare(`${right.entrypoint}\0${right.name}`));

process.stdout.write(`${JSON.stringify({ schema: "auths.docs.typescript-surface/1", package: "@auths-dev/sdk", symbols }, null, 2)}\n`);

File: bindings/python/tools/docs_surface.py
from __future__ import annotations

import json
from pathlib import Path


ROOT = Path(__file__).resolve().parents[1]


def main() -> None:
    module: str | None = None
    symbols: list[dict[str, str]] = []
    for line in (ROOT / "api" / "public-api.txt").read_text(encoding="utf-8").splitlines():
        if line.startswith("[") and line.endswith("]"):
            module = line[1:-1]
        elif line:
            if module is None:
                raise SystemExit("public API symbol has no module")
            symbols.append({"module": module, "name": line})
    symbols.sort(key=lambda symbol: (symbol["module"], symbol["name"]))
    print(json.dumps({"schema": "auths.docs.python-surface/1", "package": "auths", "symbols": symbols}, indent=2))


if __name__ == "__main__":
    main()
